### PR TITLE
Solidity array of structs allocation (mstore stress test)

### LINKED ISSUE #16: Memory benchmarks
[This](https://github.com/NomicFoundation/hardhat/issues/3316) example case is about creating Solidity memory arrays of structs. What that boils down to is a number of `MSTORE`s for [clearing out the allocated memory](https://github.com/ethereum/solidity/issues/13218) (as it could be dirty).

@jochem-brouwer has found a [pathological example](https://github.com/NomicFoundation/hardhat/issues/3316#issuecomment-1304064298) for ethereumjs, and has an [improvement for ethereumjs](https://github.com/ethereumjs/ethereumjs-monorepo/pull/2405).

### PR DESCRIPTION
**Pull request by rodiazet**
This example case (https://github.com/NomicFoundation/hardhat/issues/3316) is about creating
Solidity memory arrays of structs. What that boils down to is a number of MSTOREs 
for clearing out the allocated memory (https://github.com/ethereum/solidity/issues/13218) (as it could be dirty).

Closes #16

### CODE REVIEW

**Review by chfast — COMMENTED** [review]
This looks and works good. But I'm not super happy about the name. Maybe `array_alloc` or `structarray_alloc`?

**Comment by rodiazet** [comment]
> structarray_alloc

Done.

**Comment by rodiazet** [comment]
> This looks and works good. But I'm not super happy about the name. Maybe `array_alloc` or `structarray_alloc`?

Done

File: src/main/structarray_alloc.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+structarray_alloc:
```
Comment (chfast) [inline src/main/structarray_alloc.yml:5]: Please put some information where do we have it from.

**Pull request merged**